### ISSUE TITLE
Does dapo currently support to use reward model?

### ISSUE THREAD

**Issue by zpcalan** [verl-project/verl#5020]
Recently I'm using dapo to run rl training. But when I check the `recipe/dapo` dir, I find out that no `reward_model.model.path` is set in the scripts. And when I check the code, it says:
```python
# perform validation before training
        # currently, we only support validation using the reward_function.
        if self.val_reward_fn is not None and self.config.trainer.get("val_before_train", True):
            val_metrics = self._validate()
            assert val_metrics, f"{val_metrics=}"
            pprint(f"Initial validation metrics: {val_metrics}")
            logger.log(data=val_metrics, step=self.global_steps)
            if self.config.trainer.get("val_only", False):
                return
```
only support reward function. Does this mean:
1. If I want to use dapo and validate during training, I must use reward function instead of my reward model?
2. If I have to use my reward model, does dapo trainer support validating during training. As far as I understand, IT DOES NOT SUPPORT. Do I have to set trainer.val_before_train to False if using reward model?

Thank you for your excellant job. Hope to receive your response! : )

**Comment by Kirrito-k423** [comment]
Good questions about DAPO reward model support. Let me clarify:

**1. Current DAPO reward support:**
DAPO recipe is designed to work with reward **functions** (like GSM8K answer verification), not reward **models**. The validation path (`_validate`) only supports `val_reward_fn`.

**2. Using reward models with DAPO:**
Yes, you would need to set `trainer.val_before_train=False` if you want to use a reward model. The current DAPO trainer doesn't support validation with reward models during training.

**3. Workarounds:**

Option A: Create a reward function that calls your reward model:
```python
def reward_fn(data_source, solution_str, ground_truth, **kwargs):
    # Load your reward model
    with torch.no_grad():
        score = reward_model.score(solution_str)
    return score
```

Option B: Use the standard PPO trainer (`verl.trainer.main_ppo`) which supports reward models, instead of DAPO trainer.

**4. Feature request:**
If you need DAPO + reward model validation, consider contributing a PR that adds `val_reward_model_path` config support. The implementation would involve:
- Loading the reward model in the trainer
- Running inference during validation
- Logging reward model metrics

This would be a valuable addition for users with trained reward models.

**Comment by zpcalan** [comment]
Got it. Thanks for your detailed explanation and workaround! Issue closed.
